app/models/server/models/Rooms.js

```diff
@@ -544,24 +544,53 @@ export class Rooms extends Base {
 		return this._db.find(query, options);
 	}
 
-	findByNameOrFNameAndTypeIncludingTeamRooms(name, type, teamIds, options) {
+	findByNameOrFNameAndRoomIdsIncludingTeamRooms(name, teamIds, roomIds, options) {
 		const query = {
-			t: type,
+			t: {
+				$in: [
+					'p',
+					'c',
+				],
+			},
 			teamMain: {
 				$exists: false,
 			},
 			$and: [
 				{
 					$or: [
 						{
-							teamId: {
-								$exists: false,
-							},
+							$and: [
+								{
+									t: 'p',
+								},
+								roomIds ? { _id: { $in: roomIds } } : false,
+							],
+						},
+						{
+							$and: [
+								{
+									t: 'c',
+								},
+								{
+									teamId: {
+										$exists: false,
+									},
+								},
+							],
 						},
 						{
-							teamId: {
-								$in: teamIds,
-							},
+							$and: [
+								{
+									teamId: {
+										$exists: true,
```

Comment:
Here, you can combine both `$exists` and `$in` in the same object
```
teamId: {
   $exists: true,
   $in: ids,
}
```

Reply:
actually, `$exists: true` is not required, since it will be one of the list from `$in` statement it is implicit that it should exists =)

Reply:
Good suggestion 😄 ! I've just replaced it by
```
teamId: {
   $in: ids,
}
```